### PR TITLE
send invitations for shared calendars

### LINKED ISSUE #3830: No calendar invitation send for event in shared calendars
Hello, i can´t invite people via e-mail to my appointments out of thunderbird/lightning. I can´t invite people with the android app davdroid.
Also no invitation with the browser, chrome 57.0.2987.98 and firefox 52.0
### Steps to reproduce
1. create appointment
2. enter adresses for invitation
3. save

### Expected behaviour
thunderbird created a mesage if i would like to send the invitation
thinderbird sends invitation

### Actual behaviour
nothing happens after saving of new appointment, and no message is sent
in the window for new appointament in "invitation" the field for "send invitation" is gray and can´t click.

### Server configuration
**Operating system**:
Windows 10, ubuntu 16.10, linuxmint 18.1
**Web server:**
apache 2.4.18
**Database:**
mysql 5.7.17
**PHP version:**
7.0.15
**Nextcloud version:** (see Nextcloud admin page)
nextcloud 11.0.2
**Updated from an older Nextcloud/ownCloud or fresh install:**
updated
**Where did you install Nextcloud from:**
https://nextcloud.com

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #3830

### CODE REVIEW

**Comment by georgehrke** [comment]
note to myself: Can this solution cause issues with infinite invitation loops? Should the address list only be limited to the current user accessing it and not to all users possibly having access to it?

**Comment by georgehrke** [comment]
> Can this solution cause issues with infinite invitation loops?

Possibly

> Should the address list only be limited to the current user accessing it and not to all users possibly having access to it?

Yes, just replace `$calendarNode->getOwner()` with `$calendarNode->getPrincipalURI()`

**Comment by georgehrke** [comment]
In contrast to the previous patch this doesn't check whether the current user is a rw-share, but then a person with a ro-share should not be able to update calendar objects and trigger the `calendarObjectChange` method anyway.

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -98,4 +102,67 @@ function propFindDefaultCalendarUrl(PropFind $propFind, INode $node) {
...
+		);
+
+		if (!$isNew) {
+			$node = $this->server->tree->getNodeForPath($request->getPath());
+			$oldObj = Reader::read($node->get());
+		} else {
+			$oldObj = null;
+		}
+
+		$this->processICalendarChange($oldObj, $vCal, $addresses, [], $modified);
```
Comment (rullzer) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php]: if new participant addresses don't have an e-mail set does this do :boom: ?

**Comment by rullzer** [comment]
Tests would be appreciated as this is something we might else easily break in the future

File: apps/dav/lib/CalDAV/Schedule/Plugin.php (hunk 1 above)
Comment (georgehrke) [reply apps/dav/lib/CalDAV/Schedule/Plugin.php]: No, this is properly handled by `processICalendarChange`.

As I said in the methods comment, this is basically just a copy of `parent:: calendarObjectChange`, i just replaced line 133. That's why I also didn't write tests 🙈

Comment (rullzer) [reply apps/dav/lib/CalDAV/Schedule/Plugin.php]: Fair enough!

**Review by rullzer — APPROVED** [review]
LGTM

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by MorrisJobke** [comment]
Reverted in #15676

**Pull request merged**